src/main/resources/rules/crime/means/calculations/calculation.drl

```diff
@@ -16,71 +17,282 @@ import com.laa.model.OtherIncome;
+			annualTax = employedIncome.getIncomeTax().multiply(new BigDecimal(26));
+		}
+		 else if (employedIncome.getFrequency() == PaymentFrequency.ANNUAL) {
+			annualPay = employedIncome.getPay();
+			annualNI = employedIncome.getNationalInsurance();
+			annualTax = employedIncome.getIncomeTax();
+		}
+		
+		$decision.setPayeAndNi(BigDecimal.ZERO.add(annualTax).add(annualNI));
+		$decision.setEmploymentIncome(annualPay.setScale(2, BigDecimal.ROUND_HALF_EVEN));
+		$decision.setGrossHouseholdIncome(Optional.ofNullable($decision.getGrossHouseholdIncome()).orElse(BigDecimal.ZERO).add(annualPay).setScale(2, BigDecimal.ROUND_HALF_EVEN));
+		$decision.setTotalOutgoings(Optional.ofNullable($decision.getTotalOutgoings()).orElse(BigDecimal.ZERO).add($decision.getPayeAndNi()));
+	 System.out.println("annualPay " + annualPay);
+		
+	 System.out.println("annualTax " + annualTax);
+	 System.out.println("annualNI " + annualNI);
+	 
+		update($decision);
+end
 
-        System.out.println("averageAnnualSalary " + averageAnnualSalary);
 
-		$decision.setEmploymentIncome(averageAnnualSalary);
-		$decision.setGrossHouseholdIncome(Optional.ofNullable($decision.getGrossHouseholdIncome()).orElse(BigDecimal.ZERO).add(averageAnnualSalary));
+rule "calculate partner income, gross household income"
+lock-on-active  true
+	when
+		$case : CrimeCase( meansInformation.partner != null &&  meansInformation.partner.employedIncome   != null)
+		$decision : CrimeMeansDecisionReport( partnerEmploymentIncome == null  )
+	then
+	
+		//todo we should consider ni and tax as well like we do for applicant, spreadsheet does not have such case.
+		EmployedIncome employedIncome  = $case.getMeansInformation().getPartner().getEmployedIncome();
+		
+		BigDecimal annualPay = new BigDecimal(0.00);
+		
+		if (employedIncome.getFrequency() == PaymentFrequency.MONTHLY) {
+			annualPay = employedIncome.getPay().multiply(new BigDecimal(12));
+			
+		} else if (employedIncome.getFrequency() == PaymentFrequency.WEEKLY) {
+			annualPay = employedIncome.getPay().multiply(new BigDecimal(52));
+			
+		}
+		else if (employedIncome.getFrequency() == PaymentFrequency.BIWEEKLY) {
+			annualPay = employedIncome.getPay().multiply(new BigDecimal(26));
+		}
+		 else if (employedIncome.getFrequency() == PaymentFrequency.ANNUAL) {
+			annualPay = employedIncome.getPay();
+		}
+		
+		$decision.setPartnerEmploymentIncome(annualPay.setScale(2, BigDecimal.ROUND_HALF_EVEN));
+		// can we set grossHouseholdIncome to zero so that I don't do the null check below. this oncome is about addition not substration. check in other rules as well.
+		$decision.setGrossHouseholdIncome(Optional.ofNullable($decision.getGrossHouseholdIncome()).orElse(BigDecimal.ZERO).add(annualPay).setScale(2, BigDecimal.ROUND_HALF_EVEN));
+		
+	 	System.out.println("partner annualPay " + annualPay);
+		
+		update($decision);
+end
 
+rule "process applicant's child benefit, gross household income "
+lock-on-active true
```

Comment:
Coming to think about it i think if we asociate a pament frequency to a muliplier i..e monthly to 12 weekly to 52 you can simplify the whole lot